kernel/main.c

```diff
@@ -359,6 +418,9 @@ kmain(u32 multiboot_magic, u32 mbi_addr)
    init_kmalloc();
    init_paging();
 
+   if (booting_with_uefi)
```

Comment:
Cosmetic, but too ugly: don't check for this boolean here. Call the function unconditionally and check the boolean there (and return if it's false).